### PR TITLE
Convert `PDFPageViewBuffer` to a standard class, and use a `Set` internally

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Please refer to the individual commit messages.*

### CODE REVIEW

File: web/base_viewer.js [hunk 1]
```diff
@@ -112,31 +131,38 @@ function PDFPageViewBuffer(size) {
...
+      const ii = buf.size;
+      let i = 1;
+      for (const view of buf) {
+        if (idsToKeep.has(view.id)) {
+          buf.delete(view); // Move the view to the "end" of the buffer.
+          buf.add(view);
+        }
+        if (++i > ii) {
+          break;
+        }
```
Comment (Snuffleupagus) [inline web/base_viewer.js:148]: *Please note:* The unit-test was updated to ensure that I didn't mess up this particular condition.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
This looks great; thank you for doing this!

**Pull request merged**